### PR TITLE
[API-5329] Yehor/api 5329 added bs proxy support

### PR DESCRIPTION
**Pull request by nimnull**
## RATIONALE

Brought back env proxy settings support since we use overrided flow present in requests.request method.

Also uncovered and fixed issue when BS raises an exception on 0 requests made

### CODE REVIEW

**Comment by nimnull** [comment]
@Axik case added

File: datarobot_batch_scoring/network/network.py [hunk 1]
```diff
@@ -106,7 +109,8 @@ def _request(self, request):
 
         try:
             prepared = self.session.prepare_request(request)
-            self.session.send(prepared, timeout=self._timeout)
+            proxies = get_environ_proxies(prepared.url)
```
Comment (doleks) [inline datarobot_batch_scoring/network/network.py]: Can you add a comment to the documentation explaining that we are trying to use the right proxy settings? Also tell that it still does not work on Windows, if the proxy is set up using Auto-config https://en.wikipedia.org/wiki/Proxy_auto-config`?

Comment (doleks) [reply datarobot_batch_scoring/network/network.py]: Despite being truly ugly technology, the auto-config is wildly popular in the enterprises. I keep encountering it with the DataRobot customers.

Comment (nimnull) [reply datarobot_batch_scoring/network/network.py]: @doleks I agree with it nut let's create a separate ticket to implement windows autoconfig support.

Comment (doleks) [reply datarobot_batch_scoring/network/network.py]: For now, I suggest that we just document in the user-visible .readme that batch scoring will try to use proxy and document the clear case, when it does not work.

**Pull request merged**